azurelinuxagent/ga/agent_update_handler.py

```diff
@@ -322,49 +318,46 @@ def run(self, goal_state):
                 if "Missing requested version" in GAUpdateReportState.report_error_msg:
                     GAUpdateReportState.report_error_msg = ""
 
-            if requested_version == CURRENT_VERSION:
+            # Check if an update is allowed and update next upgrade times even if no new agent is available for upgrade
+            if not self.__check_if_agent_update_allowed_and_update_next_upgrade_times(requested_version):
                 return
 
-            # Check if an update is allowed
-            if not self.__should_update_agent(requested_version):
+            if requested_version == CURRENT_VERSION:
                 return
 
             if warn_msg != "":
                 self.__log_event(LogLevel.WARNING, warn_msg)
 
-            try:
-                # Downgrades are not allowed for self-update version
-                # Added it in try block after agent update timewindow check so that we don't log it too frequently
-                if not self.__check_if_downgrade_is_requested_and_allowed(requested_version):
-                    return
-
-                daemon_version = self.__get_daemon_version_for_update()
-                if requested_version < daemon_version:
-                    # Don't process the update if the requested version is less than daemon version,
-                    # as historically we don't support downgrades below daemon versions. So daemon will not pickup that requested version rather start with
-                    # installed latest version again. When that happens agent go into loop of downloading the requested version, exiting and start again with same version.
-                    #
-                    raise AgentUpdateError("The Agent received a request to downgrade to version {0}, but downgrading to a version less than "
-                                           "the Agent installed on the image ({1}) is not supported. Skipping downgrade.".format(requested_version, daemon_version))
-
-                msg = "Goal state {0} is requesting a new agent version {1}, will update the agent before processing the goal state.".format(
-                    self._gs_id, str(requested_version))
-                self.__log_event(LogLevel.INFO, msg)
-
-                agent = self.__download_and_get_agent(goal_state, agent_family, agent_manifest, requested_version)
+            # Downgrades are not allowed for self-update version
+            # Added it in try block after agent update timewindow check so that we don't log it too frequently
```

Comment:
try block is gone

Reply:
That was added because I need finally block before. Now I put that finallly logic when check the times, So I don't need finally, so I removed try too.

Reply:
so what does that comment mean? can you clarify it?

Reply:
I should remove this msg